### PR TITLE
Fix compile warnings

### PR DESCRIPTION
**Pull request by vorot93**

### CODE REVIEW

**Comment by ordian** [comment]
> LGTM, but I'm not sure what all the angry "GitHub Actions" warnings/failures are about.

Seems like a legit compilation error https://github.com/openethereum/openethereum/pull/11595/checks?check_run_id=553224932.

**Comment by vorot93** [comment]
It appears that the structs as a whole are always used but the fields inside are not. I added conditional dead code ignore.

File: rpc/src/v1/types/derivation.rs [hunk 1]
```diff
@@ -30,16 +30,16 @@ pub enum DerivationType {
 }
 
 /// Derivation request by hash
-#[cfg(any(test, feature = "accounts"))]
+#[cfg_attr(not(any(test, feature = "accounts")), allow_unused)]
```
Comment (ordian) [inline rpc/src/v1/types/derivation.rs:33]: you meant `allow(unused)`?

Comment (vorot93) [reply rpc/src/v1/types/derivation.rs]: Morning brain fart, sorry. :-)

File: parity/account_utils.rs [hunk 2]
```diff
@@ -48,11 +48,11 @@ mod accounts {
 		Ok(None)
 	}
 
-	pub fn private_tx_signer(_account_provider: Arc<AccountProvider>, _passwords: &[Password]) -> Result<Arc<::ethcore_private_tx::Signer>, String> {
+	pub fn private_tx_signer(_account_provider: Arc<dyn AccountProvider>, _passwords: &[Password]) -> Result<Arc<::ethcore_private_tx::Signer>, String> {
```
Comment (ordian) [inline parity/account_utils.rs:51]: but.. `AccountProvider` *is* a struct

**Pull request merged**